redbot/cogs/admin/converters.py

```diff
@@ -7,15 +7,33 @@
 
 class SelfRole(commands.Converter):
     async def convert(self, ctx: commands.Context, arg: str) -> discord.Role:
-        admin = ctx.command.cog
+        admin = ctx.command.cog.bot.get_cog("Admin")
         if admin is None:
             raise commands.BadArgument(_("The Admin cog is not loaded."))
 
+        selfroles = await admin.config.guild(ctx.guild).selfroles()
         role_converter = commands.RoleConverter()
-        role = await role_converter.convert(ctx, arg)
 
-        selfroles = await admin.config.guild(ctx.guild).selfroles()
+        pool = []
+        for role_id in selfroles:
+            role = ctx.guild.get_role(role_id)
+            if role is None:
+                continue
+            if role.name.lower() == arg.lower():
```

Comment:
What if the argument is a role ID, you didn't account for that here

Reply:
I did.  It would catch in the if `len(pool) == 0` suite.  Unless there's a role that happens to be named something like `355573989923487765`, which I would think should have priority over a role with that as its ID.  That's a VERY edge case though.